pkg/netpol/connlist/exposure_analysis_test.go

```diff
@@ -0,0 +1,192 @@
+				lenEgressExposedConns:  0,
+			},
+			// workload 2 is not protected at all (unsecure exposed)
+			wl2ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     false,
+				isEgressProtected:      false,
+				lenIngressExposedConns: 0,
+				lenEgressExposedConns:  0,
+			},
+		},
+		{
+			testName:                       "test_only_matched_rules",
+			expectedNumRepresentativePeers: 0,
+			expectedLenOfExposedPeerList:   1,
+			// workload 1 is protected and connected with known namespace in the cluster on both directions (exposed securely)
+			// workload 2 is not protected at all (unsecure exposed)
+			wl2ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     false,
+				isEgressProtected:      false,
+				lenIngressExposedConns: 0,
+				lenEgressExposedConns:  0,
+			},
+		},
+		{
+			testName:                       "test_multiple_unmatched_rules",
+			expectedNumRepresentativePeers: 3,
+			expectedLenOfExposedPeerList:   1,
+			// workload 1 is protected by ingress netpol but exposed to unknown namespaces; not protected on egress
+			wl1ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     true,
+				isEgressProtected:      false,
+				lenIngressExposedConns: 3,
+				lenEgressExposedConns:  0,
+			},
+		},
+		{
+			testName:                       "test_same_unmatched_rule_in_ingress_egress",
+			expectedNumRepresentativePeers: 1,
+			expectedLenOfExposedPeerList:   1,
+			// workload 1 is exposed unsecure to same unknown namespace on both directions
+			wl1ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     true,
+				isEgressProtected:      true,
+				lenIngressExposedConns: 2, // one to the unmatched rule and other to the entire cluster (no containment)
+				lenEgressExposedConns:  1,
+			},
+		},
+		{
+			testName:                       "test_with_no_netpols",
+			numOfParsingErrs:               1, // no netpols
+			expectedNumRepresentativePeers: 0,
+			expectedLenOfExposedPeerList:   1,
+			// workload 1 is not protected by any netpol
+			wl1ExpDataInfo: expectedPeerResultInfo{
+				isIngressProtected:     false,
+				isEgressProtected:      false,
+				lenIngressExposedConns: 0,
+				lenEgressExposedConns:  0,
+			},
+		},
+	}
+	for _, tt := range cases {
+		tt := tt
+		t.Run(tt.testName, func(t *testing.T) {
+			t.Parallel()
+			pe := buildPolicyEngineWithExposureAnalysis(t, tt.testName, tt.numOfParsingErrs)
+			// get real peers
+			peerList, err := pe.GetPeersList()
+			require.Nil(t, err, "test %q: error getting peer list", tt.testName)
+			representativePeers := pe.GetRepresentativePeersList()
+			require.Equal(t, tt.expectedNumRepresentativePeers, len(representativePeers),
+				"test %q: mismatch in number of representative peers", tt.testName)
+			peerList = append(peerList, representativePeers...)
+			peers := convertEvalPeersToConnlistPeer(peerList)
+			ca := NewConnlistAnalyzer(WithExposureAnalysis())
+			_, exposureMap, err := ca.getConnectionsBetweenPeers(pe, peers)
```

Comment:
not sure I follow.. the functions from the main flow of `connlist` cannot be used for testing? 
why do we need dedicated code here, that "kind of " repeats the same flow from the main API?

Reply:
ok now I got the question, the main API funcs can be used but I thought that since this test tests only "exposure-analysis" functionality, I should use the relevant funcs only.
will replace the calls with `ca` calls

Reply:
done